src/main/java/SaveToFile.java

```diff
@@ -0,0 +1,65 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+public class SaveToFile {
+    /**
+     * Saves tasks to out.txt.
+     *
+     * @param list TaskList to write to out.txt.
+     * @throws IOException If file not available.
+     */
+    public static void usingFileWriter(String list) throws IOException {
+        FileWriter fileWriter = new FileWriter("./out.txt");
+        fileWriter.write(list);
+        fileWriter.close();
+    }
+
+    /**
+     * Loads saved TaskList to Duke.
+     *
+     * @param filePath File path for duke's task list.
+     * @return ArrayList of saved TaskList if not new ArrayList would be returned.
+     */
+    public ArrayList<Task> loadList(String filePath) {
+        ArrayList<Task> arr = new ArrayList<>();
+        try {
+            Scanner s = new Scanner(new File(filePath));
+            while (s.hasNext()) {
+                String fileInput = s.nextLine();
+                String[] fileStringArr = fileInput.split("\\[");
+                if (fileStringArr[1].equalsIgnoreCase("T]")) {
+                    String[] todoSplit = fileStringArr[2].split(" ");
+                    Todo readTodo = new Todo(todoSplit[1]);
+                    readTodo.isDone(todoSplit[0]);
+                    arr.add(readTodo);
+                } else if (fileStringArr[1].equalsIgnoreCase("D]")) {
```

Comment:
The cases for D and E are similar to a certain extent so you might want to consider refactoring this part. For example you could make a general parser method that calls different parsing methods based on what kind of task it is, to make your code more organised.